### PR TITLE
Extract list of dependencies from the BOM

### PR DESCRIPTION
**Pull request by gaeljw**
Solves #9 

This adds:
- a `bomDependencies: Seq[ModuleID]` method to `Bom` instances
- a `Bom.dependencies(...)` helper method in the companion object for cases where you only care about `bomDependencies` but don't need to extract anything else manually from the `Bom`

I've also updated the README with documentation on this new usage and small changes (each in a different commit).

### Tests

I've tested this locally using the example described in #9 and it works successfully.

Example `build.sbt` file:
```scala
import sbt._
import com.here.bom.Bom

lazy val jacksonBom = Bom.dependencies("com.fasterxml.jackson" % "jackson-bom" % "2.16.0")

lazy val `sbt-bom-example-github` = project
  .in(file("."))
  .settings(jacksonBom)
  .settings(
    name := "sbt-bom-example-github",
    // resolvers := Resolver.mavenCentral +: resolvers.value,
    libraryDependencies += "com.typesafe.play" %% "play" % "2.8.21",
    libraryDependencies += "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.15.0",
    dependencyOverrides ++= jacksonBom.key.value,
  )
```

Which gives the expected result in terms of versions. Here's a `sbt dependencyTree` extract:
```
[info] sbt-bom-example-github:sbt-bom-example-github_2.12:0.1.0-SNAPSHOT [S]
[info]   +-com.fasterxml.jackson.module:jackson-module-scala_2.12:2.16.0
[info]   +-com.typesafe.play:play_2.12:2.8.21 [S]
[info]     +-com.fasterxml.jackson.core:jackson-annotations:2.16.0
[info]     +-com.fasterxml.jackson.core:jackson-core:2.16.0
[info]     +-com.fasterxml.jackson.core:jackson-databind:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-annotations:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-core:2.16.0
[info]     | 
[info]     +-com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-core:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-databind:2.16.0
[info]     |   +-com.fasterxml.jackson.core:jackson-annotations:2.16.0
[info]     |   +-com.fasterxml.jackson.core:jackson-core:2.16.0
[info]     |   
[info]     +-com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-annotations:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-core:2.16.0
[info]     | +-com.fasterxml.jackson.core:jackson-databind:2.16.0
[info]     |   +-com.fasterxml.jackson.core:jackson-annotations:2.16.0
[info]     |   +-com.fasterxml.jackson.core:jackson-core:2.16.0
...
```

### CODE REVIEW

File: plugin/src/test/scala/com/here/bom/internal/BomReaderSpec.scala [hunk 1]
```diff
@@ -70,9 +74,16 @@ class BomReaderSpec extends AnyFlatSpec with Matchers with MockFactory {
 
     val dependency = "com.test" %% "test_module" % bom
     dependency.revision should be("1.0")
+
+    bom.bomDependencies shouldBe Seq("com.test" % "test_module_2.12" % "1.0")
   }
 
-  private def createMockPomFile(groupId: String, artifactId: String, version: String, addParent: Boolean): File = {
+  private def createMockPomFile(
```
Comment (gaeljw) [inline plugin/src/test/scala/com/here/bom/internal/BomReaderSpec.scala:81]: This is caused by scalafmt, I could revert it if you prefer as this is not related to any of my change.

**Comment by molekyla** [comment]
@gaeljw Sorry for delay. I haven't enough time to test this changes. At the first glance it looks good, but I want to test it on my small project. Will give take a look in a next few day

**Review by molekyla — APPROVED** [review]
Thanks for contribution. I see no issues related to introduced functionality

**Comment by gaeljw** [comment]
Thanks @molekyla . Do you know when this can be part of a release?

**Comment by molekyla** [comment]
@gaeljw  Already been released under the 1.0.8 version. Despite that label in the README.md file shows that latest version is 1.0.7

**Comment by gaeljw** [comment]
Awesome! Will use it right now then :D
